benchmarks/cholesky.bril

```diff
@@ -0,0 +1,335 @@
+  dstptr: ptr<float> = ptradd output dstidx;
+  value: float =  load srcptr;
+  store dstptr value;
+.col.loop_end:
+  col: int = add col one;
+  jmp .col.loop;
+.col.done:
+.row.loop_end:
+  row: int = add row one;
+  jmp .row.loop;
+.row.done:
+  ret;
+}
+
+# Square root function
+@sqrt(input: float) : float {
+  n: float = id input;
+  v1: float = const 0.00001;
+  precision: float = id v1;
+  v2: float = id n;
+  x: float = id v2;
+  v3: bool = const true;
+  notdone: bool = id v3;
+.for.cond.4:
+  v5: bool = id notdone;
+  br v5 .for.body.4 .for.end.4;
+.for.body.4:
+  v6: float = id n;
+  v7: float = id x;
+  v8: float = fdiv v6 v7;
+  root: float = id v8;
+  v9: float = id x;
+  v10: float = id root;
+  v11: float = fadd v9 v10;
+  root: float = id v11;
+  v12: float = const 0.5;
+  v13: float = id root;
+  v14: float = fmul v12 v13;
+  root: float = id v14;
+  v15: float = id root;
+  v16: float = id x;
+  v17: float = fsub v15 v16;
+  diff: float = id v17;
+  v19: float = id diff;
+  v20: float = const 0;
+  v21: bool = flt v19 v20;
+  br v21 .then.18 .else.18;
+.then.18:
+  v22: float = const 0;
+  v23: float = id diff;
+  v24: float = fsub v22 v23;
+  diff: float = id v24;
+  jmp .endif.18;
+.else.18:
+.endif.18:
+  v26: float = id diff;
+  v27: float = id precision;
+  v28: bool = flt v26 v27;
+  br v28 .then.25 .else.25;
+.then.25:
+  v29: bool = const false;
+  notdone: bool = id v29;
+  jmp .endif.25;
+.else.25:
+.endif.25:
+  v30: float = id root;
+  x: float = id v30;
+  jmp .for.cond.4;
+.for.end.4:
+  ret x;
+}
+
+# Cholesky decomposition
+@cholesky(size: int, arr1: ptr<float>, arr2: ptr<float>) {
+  one: int = const 1;
+  i: int = const 0;
+.i.loop:
+  condi: bool = lt i size;
+  br condi .i.body .i.done;
+.i.body:
+  j: int = const 0;
+.j.loop:
+  condj: bool = le j i;
+  br condj .j.body .j.done;
+.j.body:
+  k: int = const 0;
+.k.loop:
+  condk: bool = lt k j;
+  br condk .k.body .k.done;
+.k.body:
+  # prepare indices
+  ik_index: int = mul i size;
+  ik_index: int = add ik_index k;
+  jk_index: int = mul j size;
+  jk_index: int = add jk_index k;
+  ij_index: int = mul i size;
+  ij_index: int = add ij_index j;
+  # load values
+  ik_ptr: ptr<float> = ptradd arr2 ik_index;
+  b_ik: float = load ik_ptr;
+  jk_ptr: ptr<float> = ptradd arr2 jk_index;
+  b_jk: float = load jk_ptr;
+  ij_ptr: ptr<float> = ptradd arr1 ij_index;
+  a_ij: float = load ij_ptr;
+  value: float = fmul b_ik b_jk;
+  value: float = fsub a_ij value;
+  store ij_ptr value;
+.k.loop_end:
+  k: int = add k one;
+  jmp .k.loop;
+.k.done:
+  # j loop body continues
+  ij_index: int = mul i size;
+  ij_index: int = add ij_index j;
+  jj_index: int = mul j size;
+  jj_index: int = add jj_index j;
+  a_ij_ptr: ptr<float> = ptradd arr1 ij_index;
+  b_ij_ptr: ptr<float> = ptradd arr2 ij_index;
+  jj_ptr: ptr<float> = ptradd arr2 jj_index;
+  a_ij = load a_ij_ptr;
+  b_ij = load b_ij_ptr;
+  b_jj = load jj_ptr;
+  value = fdiv a_ij b_jj;
```

Comment:
`:float` ?

Reply:
This issue is also fixed in PR https://github.com/sampsyo/bril/pull/158